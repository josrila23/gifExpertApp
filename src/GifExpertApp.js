import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories = ["One Piece"] }) => {
  // const [categories, setCategories] = useState(["One Piece"]);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;

/*
1. Enzyme
2.Enzyme-to-json
3. Debe mostrar el componente correctamente
  *shallow
  * wrapper .toMatchSnapshot()
*/
