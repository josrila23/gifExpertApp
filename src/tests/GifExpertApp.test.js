import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Probando el <GifExpertApp />", () => {
  const wrapper = shallow(<GifExpertApp />);

  test("Debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Debe mostrar una lista de categorÃ­as", () => {
    const categories = ["One Piece", "Dragon ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
