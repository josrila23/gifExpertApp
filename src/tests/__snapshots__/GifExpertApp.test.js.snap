// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el <GifExpertApp /> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Probando el <GifExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;
